### DISCUSSION TITLE
How to configure automatic indent in python-mode

### DISCUSSION THREAD

**Discussion by synic** [syl20bnr/spacemacs#15391] [Q&A]
python-mode automatic indentation makes a lot of assumptions, it seems, based on style. How can I configure the way it indents in certain cases, or alternatively, make it indent like vim.

For example, if I have the following function:

```python
def some_function(
    arg1,
    arg2,
):
    pass
```

If I go to the end of `arg2` line, press enter, and type `arg3,`, this is what I will get:

```python
def some_function(
    arg1,
    arg2,
        arg3,
):
    pass
```

What I want is:

```python
def some_function(
    arg1,
    arg2,
    arg3,
):
    pass
````

How can I make it do what I want here?

I've tried `(setq indent-line-function 'indent-relative)`, which sort of does what I want, except if I hit enter twice, the cursor will be at the begging of the line (it seems like it only examines the last line to determine relativity).

In vim, I can type `def some_function(` and press enter, it will automatically indent 4 spaces. If I type enter twice, it will still be indented 4 spaces. This is what I'd like most.

Thanks in advance!

**Comment by Trapez** [comment]
I have noticed the same problem, but since I have python-formatter set to black and python-format-on-save set to true it all ends up properly formatted when i save.
